examples/MLIRGEMM/gemm.mlir

```diff
@@ -0,0 +1,289 @@
+                    %5 =  arith.mulf %2, %3 : f64
+                    %6 = arith.addf %4, %5 : f64
+                    affine.store %6, %C[%arg6 * 4 + 0, %arg5 * 8 + %arg8] : memref<?x?xf64>
+                    // 2
+                    %7 = affine.load %0[%arg4 * -64 + %arg6 * 4 + 1, %arg7] : memref<?x?xf64>
+                    %8 = affine.load %C[%arg6 * 4 + 1, %arg5 * 8 + %arg8] : memref<?x?xf64>
+                    %9 =  arith.mulf %3, %7 : f64
+                    %10 = arith.addf %8, %9 : f64
+                    affine.store %10, %C[%arg6 * 4 + 1, %arg5 * 8 + %arg8] : memref<?x?xf64>
+                    // 3
+                    %11 = affine.load %0[%arg4 * -64 + %arg6 * 4 + 2, %arg7] : memref<?x?xf64>
+                    %12 = affine.load %C[%arg6 * 4 + 2, %arg5 * 8 + %arg8] : memref<?x?xf64>
+                    %13 =  arith.mulf %3, %11 : f64
+                    %14 = arith.addf %12, %13 : f64
+                    affine.store %14, %C[%arg6 * 4 + 2, %arg5 * 8 + %arg8] : memref<?x?xf64>
+                    // 4
+                    %15 = affine.load %0[%arg4 * -64 + %arg6 * 4 + 3, %arg7] : memref<?x?xf64>
+                    %16 = affine.load %C[%arg6 * 4 + 3, %arg5 * 8 + %arg8] : memref<?x?xf64>
+                    %17 =  arith.mulf %3, %15 : f64
+                    %18 = arith.addf %15, %16 : f64
+                    affine.store %14, %C[%arg6 * 4 + 3, %arg5 * 8 + %arg8] : memref<?x?xf64>
+                  }
+                }
+              }
+              memref.dealloc %1 : memref<?x?xf64>
+            }
+          memref.dealloc %0 : memref<?x?xf64>
+          }
+      }
+    return 
+  }
+
+    func.func @main(){
+       // Set up dims.
+       %cM = arith.constant 2048 : index
+       %cN = arith.constant 2048 : index
+       %cK = arith.constant 2048 : index
+
+       // Set Init Value.
+       %cf1 = arith.constant 1.0 : f64
+       %cf2 = arith.constant 0.0 : f64
+       %A = memref.alloc(%cM, %cK) : memref<?x?xf64>
+       %B = memref.alloc(%cK, %cN) : memref<?x?xf64>
+
+       
+       %C = memref.alloc(%cM, %cN) : memref<?x?xf64>
+       %C1 = memref.alloc(%cM, %cN) : memref<?x?xf64>
+       %C2 = memref.alloc(%cM, %cN) : memref<?x?xf64>
+
+
+       linalg.fill
+        ins(%cf1 : f64)
+       outs(%A:memref<?x?xf64>)
+
+       linalg.fill
+        ins(%cf1 : f64)
+       outs(%B:memref<?x?xf64>)
+
+       linalg.fill
+        ins(%cf2 : f64)
+       outs(%C:memref<?x?xf64>)
+
+       linalg.fill
+        ins(%cf2 : f64)
+       outs(%C1:memref<?x?xf64>)
+
+       linalg.fill
+        ins(%cf2 : f64)
+       outs(%C2:memref<?x?xf64>)
+
+       %print_A = memref.cast %A : memref<?x?xf64> to memref<*xf64>
+       //call @printMemreff64(%print_A) : (memref<*xf64>) -> ()
+
+       %print_B = memref.cast %B : memref<?x?xf64> to memref<*xf64>
+       //call @printMemreff64(%print_B) : (memref<*xf64>) -> ()
+
+       //call @matmul1(%A, %B, %C1) : (memref<?x?xf64>, memref<?x?xf64>, memref<?x?xf64>) -> ()
```

Comment:
I would recommend splitting these functions into different files, so it will be way more convenience to test different situations using different make targets, instead of manually changing the comment here.

Reply:
OK, I will split these function into different files.